.github/workflows/ct-tests.yml

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: Constant-time tests
+permissions:
+  contents: read
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  check-ct:
+    name: CT test ${{ matrix.nix-shell }} ${{ matrix.system }} 
+    strategy:
+      fail-fast: false
+      matrix:
+        system: [ubuntu-latest, ubuntu-24.04-arm]
+        nix-shell:
+          - ci_clang14
+          - ci_clang15
+          - ci_clang16
+          - ci_clang17
+          - ci_clang18
+          - ci_clang19
+          - ci_gcc48
+          - ci_gcc49
+          - ci_gcc7
+          - ci_gcc11
+          - ci_gcc12
+          - ci_gcc13
+          - ci_gcc14
+    runs-on: ${{ matrix.system }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Setup nix
+        uses: ./.github/actions/setup-shell
+        with:
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+          nix-shell: ${{ matrix.nix-shell }}
+      - name:  Build and run test (-Os)
+        uses: ./.github/actions/ct-test
+        with:
+          cflags: -Os
```

Comment:
Is it worth having `-Oz` as well? From https://clang.llvm.org/docs/CommandGuide/clang.html:
> [-Oz](https://clang.llvm.org/docs/CommandGuide/clang.html#cmdoption-O0) Like [-Os](https://clang.llvm.org/docs/CommandGuide/clang.html#cmdoption-O0) (and thus [-O2](https://clang.llvm.org/docs/CommandGuide/clang.html#cmdoption-O0)), but reduces code size further.

Reply:
added.